cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -1548,6 +1568,11 @@ public Map<ProgramRunId, RunRecordDetail> getRuns(ApplicationId applicationId, f
   // JIRA https://issues.cask.co/browse/CDAP-2172
   @Nullable
   public RunRecordDetail getRun(ProgramRunId programRun) throws IOException {
+    try {
+      programRun = getLatestProgramRunId(programRun);
```

Comment:
This is actually not necessary with my latest change, `ProgramRunId` is now ignoring version. Could you pls rebase to `develop` branch to get the latest code also?